metrics/details/init.lua

```diff
@@ -14,21 +14,26 @@ Registry.__index = Registry
 function Registry.new()
     local obj = {}
     setmetatable(obj, Registry)
+
     obj.collectors = {}
     obj.callbacks = {}
     return obj
 end
 
 function Registry:register(collector)
-    if self.collectors[collector.name] ~= nil then
-        return self.collectors[collector.name]
+    for _, c in ipairs(self.collectors) do
+        if c.name == collector.name and c.collector == collector.collector then
```

Comment:
`c.collector == collector.colletor` looks not obvious. 
Maybe it's better to store collector type in field `type`?